syntax = "proto3";

package planning;
option go_package = "github.com/jalexanderII/zero_fintech/gen/Go/planning";

import "planning/payment_plan.proto";
import "common/payment_task.proto";

service Planning {
    // Create PaymentPlan for list of PaymentTasks
    rpc CreatePaymentPlan(CreatePaymentPlanRequest) returns (PaymentPlanResponse);
    // Update PaymentPlan
    rpc ModifyPaymentPlan(ModifyPaymentPlanRequest) returns (ModifyPaymentPlanResponse);

    //  CRUD METHODS
    rpc GetPaymentPlan(GetPaymentPlanRequest) returns (PaymentPlan);
    rpc ListPaymentPlans(ListPaymentPlanRequest) returns (ListPaymentPlanResponse);
    rpc UpdatePaymentPlan(UpdatePaymentPlanRequest) returns (PaymentPlan);
    rpc DeletePaymentPlan(DeletePaymentPlanRequest) returns (DeletePaymentPlanResponse);
}

message CreatePaymentPlanRequest {
    repeated common.PaymentTask payment_tasks = 1;
}

message PaymentPlanResponse {
    repeated planning.PaymentPlan payment_plans = 1;
}

message ModifyPaymentPlanRequest {
    repeated planning.PaymentPlan payment_plans = 1;
    MetaData meta_data = 2;
}
